storage/hasherstore.go

```diff
@@ -152,7 +152,10 @@ func (h *hasherStore) startWait(ctx context.Context) {
 		select {
 		// if context is done earlier, just return with the error
 		case <-ctx.Done():
-			h.waitC <- ctx.Err()
+			select {
+			case h.waitC <- ctx.Err():
+			case <-h.quitC:
```

Comment:
What is `quitC` ?

Reply:
check `L142` above in `func (h *hasherStore) Wait(ctx context.Context) error`

Reply:
Still not so clear, but I guess the comment by the member in the owning object suggests that this is a channel for orderly shutdown.

The description of the PR suggests a fix on a deadlock. But the code seems rather to protect against closing the `quitC` when it's already closed, which is a panic not a deadlock? This is why I was curious.

Reply:
Very well observed @nolash. Yes, this PR addresses a deadlock and a panic on closing a closed channel, problems that were found while running smoke tests on new syncer, but the fixes may be applied to the master.

Reply:
to second @janos, indeed the problem of close of closed channel is resolved by the `Once`. the change to this specific select structure is made to eliminate trivial writes to channel that are potential deadlocks